terraform_compliance/extensions/terraform.py

```diff
@@ -145,6 +147,32 @@ def _parse_resources(self):
         if self.parse_it:
             self.cache.set('resources', self.resources)
 
+    '''
+    Creates a map of resource type to after_unknown values
+    These can be used in given step "resource that supports x"
+    
+    Note: This function may be extended to capture 'after' values as well. That would require flattening the multi level
+    dictionaries in resource
+    '''
+    def remember_after_unknown(self, resource, after_unknown):
+        # get type
+        resource_type = resource.get('type', '')
+
+        # if resource doesn't have type field, try to extract it from address
+        if not resource_type and 'address' in resource and resource['address']:
+            resource_type = resource.get('address').split('.')[0]
```

Comment:
What if this is a module ?

Reply:
True, then we would most likely need to use `resource.get('address').split('.')[2]` in that case.

However, I'm inclined to delete this clause all together. Is it possible to not have `type` in resource at first place?

Reply:
Not sure, I think that depends on the provider really.

Reply:
I made the change. However, we might run into a similar issue with the resource's address convention if provider doesn't add the type of the resource to the plan.